File: java/src/main/java/com/iteso/nintendo/attacks/Psychic.java
package com.iteso.nintendo.attacks;

import com.iteso.nintendo.interfaces.Attack;

/**
 * Created by alex_rios on 10/3/19.
 */
public class Psychic implements Attack {
    /**
     * Psychic attack starting damage.
     */
    private static final int START_DAMAGE = 80;

    /**
     * Psychic attack damage.
     */
    private int damage = START_DAMAGE;

    /**
     * Perform Psychic attack.
     */
    @Override
    public String performAttack() {
        return "Psychic";
    }

    /**
     * Get attack damage.
     * @return damage.
     */
    @Override
    public int getAttackDamage() {
        return this.damage;
    }

    /**
     * Set attack damage.
     * @param damage
     */
    @Override
    public void setAttackDamage(final int damage0) {
        this.damage = damage0;
    }
}

File: python/src/main/python/Psychic.py
from Attack import *

class Psychic(Attack):

    def __init__(self, damage = 80):
        self.damage = damage

    def performAttack(self):
        return "Psychic"

    def getAttackDamage(self):
        return self.damage

    def setAttackDamage(self, damage):
        self.damage = damage